examples/server/server.cpp

```diff
@@ -164,7 +164,7 @@ struct server_slot {
     int32_t n_prompt_tokens           = 0;
     int32_t n_prompt_tokens_processed = 0;
 
-    json prompt; // can be either a string, array of strings or array of token ids
+    json prompt; // can be either a string, array of strings, array of token ids, or mixed array of strings and token ids
```

Comment:
If it is an array of strings, the code treats it properly as a multi-task. It is only in the case of a mixture of tokens and strings that it is treated as a singleton mixture. To me, this seemed like a good compromise.

I understand it may be out of the scope of this particular PR, but to be honest, this issue is pretty near and dear to me, and is the main reason I switched to using llama-server to begin with. Could I separate out the code into another PR or do you feel it has fundamental problems?

Reply:
> If it is an array of strings, the code treats it properly as a multi-task. It is only in the case of a mixture of tokens and strings that it is treated as a singleton mixture. To me, this seemed like a good compromise.

Ok I wasn't aware that we have ability to used mixed type. Before, something like `[12, 34, "abc", 56, 78],` won't work.

This must be refactor anyways, because the code is getting spaghetti after all of this "added features". What I suggest is that you can revert these change and I'll make a PR to refactor the task splitting and tokenization. Ideally, `slot.prompt` should be an array of tokens, it should not hold a string anymore.

Reply:
Understood. Thank you for being open to addressing this in the refactor. It is an extremely helpful feature for front-end software when you need to get into the nitty-gritty of prompting outside of an external chat-template and directly apply your own templates or experiment with different prompt styles for models that accept multiple formats, all directly using the `/completion` endpoint.

With the ability to mix tokens and strings in one prompt I can rapidly test different models and prompting styles without having to necessarily handle a bunch of different tokenizing setups internally in my own code or do a bunch of separate fetches to the server to tokenize.

But yes, I will remove this specific code in the next commit, thank you.

Reply:
I'm trying to refactor this part in next few days, will push a PR when it's ready